metadata_writer_nne.c

```diff
@@ -33,12 +33,64 @@
 #include "metadata_exporter_log.h"
 
 #define NNE_DEFAULT_INTERVAL_MS 15000
-#define NNE_DEFAULT_DIRECTORY   "/nne/data/"
-#define NNE_DEFAULT_PREFIX      "gps_"
-#define NNE_DEFAULT_EXTENSION   ".sdat"
+#define NNE_DEFAULT_DIRECTORY "/nne/data/"
+#define NNE_DEFAULT_GPS_PREFIX "gps_"
+#define NNE_DEFAULT_GPS_EXTENSION ".sdat"
+#define NNE_DEFAULT_METADATA_PREFIX "-metadatacollector-"
+#define NNE_DEFAULT_METADATA_EXTENSION ".json"
+
+
+struct nne_metadata_descr
+{
+    enum nne_metadata_idx idx;
+    const char *key;
+    enum nne_type type;
+    int offset; // offset to member in md_iface_event structure
+};
+
+struct nne_metadata_descr NNE_METADATA_DESCR[] = {
+    { NNE_IDX_MODE,    "mode",    NNE_TYPE_UINT8,  offsetof(struct md_iface_event, device_mode)    },
+    { NNE_IDX_SUBMODE, "submode", NNE_TYPE_UINT8,  offsetof(struct md_iface_event, device_submode) },
+    { NNE_IDX_RSSI,    "rssi",    NNE_TYPE_INT8,   offsetof(struct md_iface_event, rssi) },
+    { NNE_IDX_RSCP,    "rscp",    NNE_TYPE_INT16,  offsetof(struct md_iface_event, rscp) },
+    { NNE_IDX_ECIO,    "ecio",    NNE_TYPE_INT8,   offsetof(struct md_iface_event, ecio) },
+    { NNE_IDX_RSRP,    "rsrp",    NNE_TYPE_INT16,  offsetof(struct md_iface_event, lte_rsrp) },
+    { NNE_IDX_RSRQ,    "rsrq",    NNE_TYPE_INT8,   offsetof(struct md_iface_event, lte_rsrq) },
+    { NNE_IDX_LAC,     "lac",     NNE_TYPE_UINT16, offsetof(struct md_iface_event, lac) },
+    { NNE_IDX_CID,     "cid",     NNE_TYPE_INT32,  offsetof(struct md_iface_event, cid) },
+    { NNE_IDX_OPER,    "oper",    NNE_TYPE_UINT32, offsetof(struct md_iface_event, nw_mccmnc) }
+};
+
+#define NNE_METADATA_DESCR_LEN (sizeof(NNE_METADATA_DESCR) / sizeof(struct nne_metadata_descr))
+
+enum nne_message_type
+{
+    NNE_MESSAGE_TYPE_EVENT,
+    NNE_MESSAGE_TYPE_BINS1MIN
+};
+
+enum nne_message_source
+{
+    NNE_MESSAGE_SOURCE_REPORT,
+    NNE_MESSAGE_SOURCE_QUERY
+};
+
+struct nne_message
+{
+    enum nne_message_type  type;
+    uint64_t tstamp;
+    const char *node;
+    uint32_t mccmnc;
+    const char *key;
+    struct nne_value value;
+    const char* extra;
+    enum nne_message_source source;
+    uint64_t delta;
+};
```

Comment:
Personal opinion, but I think structs/enums/etc. should be plced in header files.